pkg/loadbalancer/target/metrics.go

```diff
@@ -0,0 +1,304 @@
+/*
+Copyright (c) 2023 Nordix Foundation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package target
+
+import (
+	"context"
+	"encoding/binary"
+	"sync"
+
+	"github.com/google/nftables"
+	"github.com/google/nftables/expr"
+	nspAPI "github.com/nordix/meridio/api/nsp/v1"
+	meridioMetrics "github.com/nordix/meridio/pkg/metrics"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+)
+
+const (
+	tableName = "meridio-metrics"
+	chainName = "target-hits"
+)
+
+type HitsMetrics struct {
+	hostname string
+	meter    metric.Meter
+	targets  map[int]*nspAPI.Target
+	table    *nftables.Table
+	chain    *nftables.Chain
+	mu       sync.Mutex
+}
+
+func NewTargetHitsMetrics(hostname string) (*HitsMetrics, error) {
+	meter := otel.GetMeterProvider().Meter(meridioMetrics.METER_NAME)
+	hm := &HitsMetrics{
+		hostname: hostname,
+		meter:    meter,
+		targets:  map[int]*nspAPI.Target{},
+	}
+
+	err := hm.init()
+	if err != nil {
+		return nil, err
+	}
+
+	return hm, nil
+}
+
+// init creates the nftables table and chain.
+func (hm *HitsMetrics) init() error {
+	conn := &nftables.Conn{}
+
+	hm.table = conn.AddTable(&nftables.Table{
+		Family: nftables.TableFamilyINet,
+		Name:   tableName,
+	})
+
+	hm.chain = conn.AddChain(&nftables.Chain{
+		Name:     chainName,
+		Table:    hm.table,
+		Type:     nftables.ChainTypeFilter,
+		Hooknum:  nftables.ChainHookPostrouting,
+		Priority: nftables.ChainPriorityRef(-500),
+	})
+
+	return conn.Flush()
+}
+
+// Register adds a target as nftables rule in the postrouting chain
+func (hm *HitsMetrics) Register(id int, target *nspAPI.Target) error {
+	hm.mu.Lock()
+	defer hm.mu.Unlock()
+
+	targetMetrics, err := hm.getMetrics()
```

Comment:
Looks like a huge waste fetching all the rules instead of the one you want to check.